src/rgw/sts-assume-role.cc

```diff
@@ -0,0 +1,145 @@
+#include <errno.h>
+#include <ctime>
+#include <regex>
+#include <boost/format.hpp>
+#include <boost/algorithm/string/replace.hpp>
+
+#include "common/errno.h"
+#include "common/Formatter.h"
+#include "common/ceph_json.h"
+#include "common/ceph_time.h"
+#include "rgw_rados.h"
+#include "auth/Crypto.h"
+#include "include/ceph_fs.h"
+
+#include "include/types.h"
+#include "rgw_string.h"
+
+#include "rgw_common.h"
+#include "rgw_tools.h"
+#include "rgw_role.h"
+#include "sts-assume-role.h"
+
+#define dout_subsys ceph_subsys_rgw
+
+namespace STS {
+  
+int Credentials::generateCredentials(CephContext* cct)
+{
+  uuid_d accessKey, secretKey;
+  char accessKeyId_str[MAX_ACCESS_KEY_LEN], secretAccessKey_str[MAX_ACCESS_KEY_LEN];
+
+  //AccessKeyId
+  accessKey.generate_random();
+  accessKey.print(accessKeyId_str);
+  accessKeyId = accessKeyId_str;
+
+  //SecretAccessKey
+  secretKey.generate_random();
+  secretKey.print(secretAccessKey_str);
+  secretAccessKey = secretAccessKey_str;
+
+  //Expiration
+  real_clock::time_point t = real_clock::now();
+
+  struct timeval tv;
+  real_clock::to_timeval(t, tv);
+  tv.tv_sec += EXPIRATION_TIME_IN_SECS;
+
+  struct tm result;
+  gmtime_r(&tv.tv_sec, &result);
```

Comment:
@aemerson should this be a ceph::clock call?

Reply:
No. This isn't getting the time, this is just parsing it out.